spark/src/test/scala/org/apache/spark/sql/delta/MergeIntoDVsSuite.scala

```diff
@@ -227,7 +227,9 @@ class MergeIntoDVsSuite extends MergeIntoDVsTests {
           tableHasDVs = true,
           targetDf = sourceDF.as("s").join(targetDFWithMetadata.as("t"), condition),
           candidateFiles = corruptedFiles,
-          condition = condition.expr
+          condition = condition.expr,
+          fileNameColumnOpt = Option(col("s._metadata.file_name")),
```

Comment:
To the reviewer. I think this is correct, but I am not 100% sure.

Reply:
For context: both side of the join have a matching metadata field now. Using `_metadata.file_name` or `_metadata.row_index` will fail now with an AMBIGUOUS_REFERENCE exception.